dynamite_sdk/objects/baselines.py

```diff
@@ -0,0 +1,208 @@
+import pandas as pd
+
+from ipaddress import ip_address
+from dateutil.parser import parse as date_parse
+
+
+class InvalidIntervalError(Exception):
+    """
+    Thrown when an unexpected raw interval format is encountered
+    """
+    def __init__(self, message):
+        """
+        :param message: A more specific error message
+        """
+        msg = "An error occurred while parsing an interval: {}".format(message)
+        super(InvalidIntervalError, self).__init__(msg)
+
+
+class Interval:
+
+    def __init__(self, raw_interval_document: dict):
+        self.raw_interval_document = raw_interval_document
+
+        self.elasticsearch_index = None                          #: The ElasticSearch index where the interval was found
+        self.originating_agent_tag = None                        #: The friendly name of the agent
+        self.forwarder_type = None                               #: Always "zeek"
+        self.event_type = None                                   #: Always "netbase"
+        self.node_ip_address = None                              #: The ip_address of the originating host (collector/agent IP)
+        self.node_hostname = None                                #: The hostname of the originating host (collector/agent hostname)
+        self.address = None                                      #: The monitored IP address
+        self.start_time = None                                   #: Start time of the observation
+        self.end_time = None                                     #: End time of the observation
+        self.interval_size = None                                #: The time between start_time and end_time
+        self.internal_port_count = None                          #: Count of unique ports communicated with internally
+        self.internal_host_count = None                          #: Count of unique hosts communicated with internally
+        self.external_port_count = None                          #: Count of unique ports communicated with externally
+        self.external_host_count = None                          #: Count of unique hosts communicated with externally
+        self.internal_client_count = None                        #: Count of unique internal clients communicating with this IP
+        self.external_client_count = None                        #: Count fo unique external clients communicating with this IP
+        self.connections_count = None                            #: Total count of connections this IP was involved in
```

Comment:
Suggest naming this attribute connection_count (instead of "connections" plural) to keep it consistent with everything else.  Nit picky but for usability sake it might be worth changing.